File: Champak in Escape Room/gemini2.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

void solve() {
    int t;
    if (!(cin >> t)) return;
    while (t--) {
        int n, m, k;
        cin >> n >> m >> k;
        vector<string> grid(n);
        for (int i = 0; i < n; ++i) {
            cin >> grid[i];
        }
        string s;
        cin >> s;
        
        int r = 0, c = 0;
        for (char move : s) {
            int nr = r, nc = c;
            if (move == 'U') nr--;
            else if (move == 'D') nr++;
            else if (move == 'L') nc--;
            else if (move == 'R') nc++;
            
            if (nr >= 0 && nr < n && nc >= 0 && nc < m && grid[nr][nc] == '.') {
                r = nr;
                c = nc;
            }
        }
        cout << r + 1 << " " << c + 1 << "\n";
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    solve();
    return 0;
}

File: Champak in Escape Room/gemini2.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    T = int(input_data[0])
    idx = 1
    out = []
    
    for _ in range(T):
        N = int(input_data[idx])
        M = int(input_data[idx+1])
        K = int(input_data[idx+2])
        idx += 3
        
        grid = input_data[idx : idx+N]
        idx += N
        
        S = input_data[idx]
        idx += 1
        
        r, c = 0, 0
        for move in S:
            nr, nc = r, c
            if move == 'U':
                nr -= 1
            elif move == 'D':
                nr += 1
            elif move == 'L':
                nc -= 1
            elif move == 'R':
                nc += 1
                
            if 0 <= nr < N and 0 <= nc < M and grid[nr][nc] == '.':
                r, c = nr, nc
                
        out.append(f"{r + 1} {c + 1}")
        
    print('\n'.join(out))

if __name__ == '__main__':
    solve()